app/models/odk/response_parser.rb

```diff
@@ -8,36 +8,10 @@ module ODK
   class ResponseParser
     attr_accessor :response, :raw_odk_xml, :files, :awaiting_media, :odk_hash
 
-    def self.duplicate?(xml, user_id)
-      checksum = compute_checksum_in_chunks(File.new(xml))
-      blobs = ActiveStorage::Blob.where(checksum: checksum)
-      blobs.each do |blob|
-        attachment = ActiveStorage::Attachment.find_by(blob_id: blob.id)
-        next if attachment.nil?
-        response = Response.find(attachment.record_id)
-        return true if response.present? && response.user.id == user_id
-      end
-      false
-    end
-
-    # ActiveStorage checksum; copied from Rails to be identical:
-    # https://github.com/rails/rails/blob/main@{2022-04-01}/activestorage/app/models/active_storage/blob.rb#L369
-    #
-    # rubocop:disable Naming/MethodParameterName, Lint/AssignmentInCondition
-    def self.compute_checksum_in_chunks(io)
-      OpenSSL::Digest.new("MD5").tap do |checksum|
-        while chunk = io.read(5.megabytes)
-          checksum << chunk
-        end
-        io.rewind
-      end.base64digest
-    end
-    # rubocop:enable Naming/MethodParameterName, Lint/AssignmentInCondition
-
     def initialize(response: nil, files: nil, awaiting_media: false)
       raise "Submissions must have a mission" if response.mission.nil?
       @response = response
-      @raw_odk_xml = files.delete(:xml_submission_file).read
+      @raw_odk_xml = files[:xml_submission_file].read
       @files = files
```

Comment:
@files is passed in to answer_parser, so by no longer deleting xml_submission_file above, answer_parser will try to attach the xml to the response as if it were a media attachment.

i can't tell if @files actually needs to include xml_submission_file for anything, maybe we can just revert the line above without issue?